### DISCUSSION TITLE
Desktop file - necessary to have one?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AppImage/AppImageKit#1256] [Q&A] [answered]
I'm using appimagetool to build an AppImage from an AppDir.

I'm actually trying to package a command line only tool, which I don't expect to be visible on the desktop.

However, having a desktop file appears mandatory.

Is it appropriate to have no desktop file in this scenario?

**Comment by TheAssassin** [comment, marked as answer]
Nope. You always need to have one. Desktop files aren't for GUI tools only. Also, tools that work with AppImages may still want to display some information, e.g., an icon in the file browser. You can set `Terminal=true` in your desktop file to signalize it's a CLI application.

**Comment by kuiperzone** [comment]
OK, I accept your answer.

But it doesn't quite sense. Let's support I integrate the desktop file, then I'll get an option to launch it from the desktop  -- and when I do it's just going to show me an error because it's missing command line arguments.

Is there anything in the desktop file itself which says, look, just don't show this?

**Reply by probonopd** [reply]
Set `Terminal=true` and `NoDisplay=true` in the desktop file if you want to signal that it is a command line application and should not be shown in the menu.

**Reply by kuiperzone** [reply]
Perfect! Just what I was looking for. Thank you.

**Reply by TheAssassin** [reply]
There also is `X-AppImage-Integrate=false`. Some tools (including AppImageLauncher) ignore `NoDisplay=true` tools as well, but it's not part of the standard. If you want to make sure people using AppImageLauncher are not prompted, set all three.
